lib/cc/analyzer/issue.rb

```diff
@@ -0,0 +1,58 @@
+module CC
+  module Analyzer
+    class Issue
+      SPEC_ISSUE_ATTRIBUTES = %w[
+        categories
+        check_name
+        content
+        description
+        location
+        other_locations
+        remediation_points
+        severity
+        type
+      ]
+
+      def initialize(output)
+        @output = output
+      end
+
+      def as_json(*)
+        parsed_output.reverse_merge!(
+          "fingerprint" => fingerprint,
+        )
+      end
+
+      def fingerprint
+        parsed_output.fetch("fingerprint", default_fingerprint)
+      end
+
+      # Allow access to hash keys as methods
+      SPEC_ISSUE_ATTRIBUTES.each do |key|
+        define_method(key) do
```

Comment:
Open to other ideas on this. I'm torn between doing this and using `method_missing`. What do others think?